### PR TITLE
feat: add outputMode 'thread' to automation agent node (RUN_AGENT)

### PR DESCRIPTION
**Pull request by XyneSpaces**
## Summary

The automation **agent node** (`RUN_AGENT` step) previously forced every agent run into **capture mode**: the agent's response was parsed into a fixed `{ result }` JSON envelope for downstream steps, and nothing was posted back to the conversation that triggered the automation. Teams that just want the agent's answer delivered into the originating thread had to add a separate `REPLY_ON_MESSAGE` step wired to `step.output.result`.

This PR adds an **`outputMode`** toggle on the node:

- **`capture`** (default, unchanged) — parse the response into `{ result }` JSON, validate against `outputSchema`, retry on mismatch. Existing behavior, byte-for-byte.
- **`thread`** — claw-auth *also* posts the agent's final answer into the bound conversation (`channelId` + `conversationId`), then still forwards the completion so the workflow step resumes. Output-schema validation and the empty-result retry are skipped; the node returns `{ delivered: true }` (plus any attachments), and downstream steps cannot read a result from it.

## How it works

| Layer | Change |
|---|---|
| `apps/backend/src/automations/steps/run-agent.step.ts` | New `outputMode` config field. `execute()` requires a conversation-bearing trigger in thread mode and passes `deliverToThread` to claw. `onResume()` short-circuits before schema capture in thread mode and returns `{ delivered: true }`. |
| `apps/backend/src/automations/services/claw-client.ts` | Threads `deliverToThread` through `RunAgentRequest` into the signed `/app/:spacesAppId` payload. |
| `apps/xyne-claw-auth/backend/src/lib/session-context.ts` | New optional `SessionContext.deliverResultToThread`. |
| `apps/xyne-claw-auth/backend/src/routes/webhook.ts` | `handleAutomationWebhook` sets `deliverResultToThread` only when a conversation is bound. The existing `/webhook/result` result-forward branch posts the resolved final answer (mentions already expanded) into the thread as the agent bot — fail-open — before forwarding the completion callback. |

`resultForwardUrl` is intentionally **kept set** in thread mode so the step still resumes reliably; this also avoids the `EXTERNAL_WAIT` hang a forward-less conversation run would cause. The thread post reuses the exact `spacesAppFetch("/chat/postMessage", ...)` shape already used elsewhere in the same handler.

## Blast radius

Additive and guarded. `outputMode` defaults to `capture`, so **existing automations are completely unaffected**. The only new behavior fires when a node is explicitly set to `thread` **and** the trigger carries a conversation.

## Testing needed (please run in CI / a real env)

- The **backend** package typechecks clean locally (`tsc --noEmit`, exit 0).
- **claw-auth** full `tsc` could not complete in the dev sandbox because `prisma generate` needs network access to fetch engine binaries (offline sandbox), which produces the pre-existing `@prisma/client has no exported member 'Prisma'` cascade unrelated to this diff. My changed lines produce **zero** type errors. Please confirm `apps/xyne-claw-auth/backend` typecheck is green in CI.
- **End-to-end not verified in-sandbox** — needs a live run: build an automation whose agent node is set to `outputMode: thread`, trigger it from a message/ticket, and confirm (a) the agent's final answer is posted into the trigger thread as the agent bot with resolved mentions, and (b) the workflow step resumes (does not hang in `EXTERNAL_WAIT`). Also verify the guard error fires when `thread` mode is used on a trigger with no conversation.

## Explicitly out of scope (follow-up)

Live token-by-token streaming into the thread and interactive mid-run question cards are **not** included — automation runs are headless (no SSE), so those need separate runtime work. This PR delivers the agent's **final** answer to the thread.

### CODE REVIEW

File: apps/backend/src/automations/steps/run-agent.step.ts [hunk 1]
```diff
@@ -118,6 +131,21 @@ export class RunAgentStep extends BaseActionStep<typeof RunAgentConfigSchema, Ru
       throw new Error(`[RUN_AGENT] claw run status=${String(envelopeStatus)}: ${String(envErr)}`);
     }
 
+    // Thread-output mode: the agent's answer was posted into the trigger
+    // conversation by claw-auth. There is no { result } JSON to validate or hand
```
Comment (XyneSpaces) [inline apps/backend/src/automations/steps/run-agent.step.ts:135]: **[🟠 High] R2** — The thread branch reports delivery even when nothing was posted: it returns `{ delivered: true }` and logs a "posted to thread" info line unconditionally, but three paths can skip or lose the post while the step still shows green:

1. **Empty final answer** — the postMessage guard in `webhook.ts` (~L5682) skips posting when the resolved answer has no non-empty text, yet this branch still returns `delivered: true`. Capture mode retries empty/invalid results via `handleValidationFailure`; thread mode skips the empty-result retry (per the PR description), so an empty answer = nothing in the thread + a successful step.
2. **Fail-open post** — the post is `spacesAppFetch("/chat/postMessage", …).catch(log.warn)`, so an expired `ctx.appToken` or any 4xx/5xx is swallowed at warn level while the step still reports delivered.
3. *(assumption, pre-existing)* — the capacity-retry block above doesn't early-return, so a "retry later" callback can fall through into this branch and be marked delivered; on the later retry a thread-mode run could post twice.

Since downstream steps can't read a result from a thread-mode node, a silently-skipped post leaves the conversation with no agent answer while the run shows success — hard to debug after the fact.

Suggested fix: report the actual post outcome back through the completion callback (e.g. a `delivered`/`postError` field populated by the real `postMessage` result rather than a fire-and-forget `.catch`), return `delivered: false` (or throw) here when the post was skipped or failed, and either route empty answers through the existing retry path or log an explicit "thread delivery skipped: empty answer" warning.

File: apps/backend/src/automations/steps/run-agent.step.ts [hunk 2]
```diff
@@ -35,6 +35,12 @@ const RunAgentConfigSchema = z.object({
     .max(10)
     .optional()
     .describe('How many times to retry the agent if its response fails schema validation. Default 3.'),
+  outputMode: z
```
Comment (XyneSpaces) [inline apps/backend/src/automations/steps/run-agent.step.ts:38]: **[🟠 High] R7** — New functionality with no tests (guidelines rule 204 — add unit tests for new functionality and edge cases). `outputMode` introduces a second delivery mode threaded through four files — the enum/default here, the `execute()` conversation guard at L84–89, the `deliverToThread` plumbing through `claw-client.ts` → `session-context.ts` → `webhook.ts`, and the `onResume()` thread branch at L135–145 — and none of it is covered by a test. The description's "Testing needed" section itself confirms the end-to-end path wasn't verified in-sandbox.

Suggested fix: add unit tests over `execute`/`onResume` covering at minimum (1) default stays `capture` and existing automations are unaffected, (2) thread mode on a trigger with no bound conversation hits the L84–89 guard error, (3) the thread-mode happy path returns `{ delivered: true }` (+ attachments), and (4) empty-answer / failed-post behavior in thread mode (see the R2 comment at L135). Per guidelines rule 208, backend feature PRs also need a proof-of-testing link — a screen recording of the live run described in the PR body would satisfy it.

**Review by XyneSpaces — COMMENTED** [review]
Automated review from **Xyne Spaces Reviewer** — 3 inline findings below (2 🟠 High, 1 🟡 Medium, 0 blocking). Full summary with the findings table and the recommended reviewer (@Dnyaneshwari-04) follows as a top-level comment on this PR.

File: apps/xyne-claw-auth/backend/src/routes/webhook.ts [hunk 3]
```diff
@@ -5666,6 +5674,32 @@ router.post("/result", requireStrictS2S, requireResultToken((req) => (req.body a
     if (payload.emptyReason === "provider_capacity" && !forwardText.trim()) {
       await scheduleCapacityRetryIfNeeded(ctx, payload, false).catch(() => false);
     }
+    // Thread-output automation node: post the agent's final answer into the
+    // conversation that triggered the automation, in addition to forwarding the
+    // completion below so the workflow step still resumes. Guarded on
+    // deliverResultToThread + a bound conversation; fail-open so a post failure
+    // never blocks the step from advancing.
+    if (
```
Comment (XyneSpaces) [inline apps/xyne-claw-auth/backend/src/routes/webhook.ts:5682]: **[🟡 Medium] R6** — New silent failure path: when this guard skips the thread post (empty resolved answer / missing conversation binding), nothing is logged, so triage can't tell "delivered" from "skipped" without querying Spaces. Suggested fix: add a `log.info`/`log.warn` next to the guard stating that thread delivery was skipped and why, so skipped deliveries are greppable.

**Comment by XyneSpaces** [comment]
## Xyne Spaces Reviewer — PR #1663

**Verdict:** Changes requested — fix the 3 🟠 High findings before merge (no 🔴 blocking issues)
**Reviewed:** 4 files, 10 hunks (+79/−0) · **Findings:** 0 blocking, 3 High, 1 Medium

### Findings
| Severity | Rule | Location | Issue |
|----------|------|----------|-------|
| 🟠 High | R2 | apps/backend/src/automations/steps/run-agent.step.ts:135 | Thread branch returns `delivered: true` and logs "posted" even when the post was skipped (empty answer) or failed (fail-open `.catch(log.warn)`) — run shows success, thread has no answer |
| 🟠 High | R5 | apps/dashboard/src/components/Automation/AutomationBuilder/StepCard/RunAgentStepForm.tsx *(not in this diff)* | Frontend drift: the Builder's `RunAgentConfigShape` (`agentSlug`/`prompt`/`outputSchema`) has no `outputMode`, so the new mode is unreachable from the Automation Builder UI |
| 🟠 High | R7 | apps/backend/src/automations/steps/run-agent.step.ts:38 | New `outputMode` behavior across 4 files with no unit tests (rule 204) and no proof-of-testing link (rule 208); E2E not verified per the description |
| 🟡 Medium | R6 | apps/xyne-claw-auth/backend/src/routes/webhook.ts:5682 | postMessage guard silently skips the thread post with no log — skipped deliveries aren't greppable |

**Notes**
- **R5 (assumption):** `maxRetries` is likewise absent from `RunAgentStepForm.tsx`, so backend-only config fields may be an accepted pattern here — if a follow-up UI PR is planned, please link it; otherwise consider shipping the `capture`/`thread` segmented control with this change (hiding `outputSchema` in thread mode, since it's ignored there).
- **R4 (assumption, pre-existing):** the capacity-retry block in `onResume` doesn't early-return, so a retried callback can fall through into the new thread branch — in thread mode that's a visible side effect (possible duplicate post on retry). Folded into the R2 inline comment; needs confirmation against the retry semantics.
- No migration files touched — migration-safety review not triggered. The `claw-client.ts` and `session-context.ts` changes are clean passthroughs.

### Recommended reviewer
**@Dnyaneshwari-04** — owns the exact code this PR modifies (`run-agent.step.ts` `execute`/`onResume`, merged in 6325b945 / #477 "allow pausing steps inside control-flow branches"; open #1657 in the same automation-steps family) and has the lowest review load of the candidates.

> ⚠️ This bot has no reviewer-request API access, so the request couldn't be filed programmatically — please add **@Dnyaneshwari-04** as a reviewer on this PR.

| Candidate | Owns (files in this PR) | Default reviewer? | Pending review requests | Tie-breaker |
|-----------|-------------------------|-------------------|-------------------------|-------------|
| @Dnyaneshwari-04 | run-agent.step.ts (execute/onResume — merged 6325b945/#477) | no — no CODEOWNERS entry covers these paths | 0 | most Spaces-active candidate (133 msgs since 25 Aug; member of #xyne-spaces-pr-review); only 2 open PRs |
| @Revanthvenkatpasupuleti | — (merged #1647, automation-trigger / RUN_AGENT path, today) | no | 0 | 7 open PRs; less Spaces activity |
| @satvikbatra | webhook.ts (open #1017 touches the same file) | no | 285 | ruled out on workload |

*Ranking basis:* repo CODEOWNERS only covers ACL paths (team `@juspay/xyne-acl-guardians`) — none of the 4 changed files have a CODEOWNER, so ranking is recent file ownership + review load + Spaces activity. PR author is the `XyneSpaces` bot (commit author: Anurag Dwivedi) — excluded, as are bot accounts. Also considered: Prajwal Prasad (#1635, claw agents fix) — less direct ownership of these paths.
